src/components/LineSeries.tsx

```diff
@@ -0,0 +1,48 @@
+import { extent } from 'd3-array';
+import { scaleLinear } from 'd3-scale';
+import { line } from 'd3-shape';
+import React, { CSSProperties, useEffect } from 'react';
+
+import { useDispatchContext, usePlotContext } from '../hooks';
+import type { LineSeriesProps, Series } from '../types';
+
+interface LineSeriesRenderProps {
+  width?: number;
+  height?: number;
+  data: Series[];
+  lineStyle?: CSSProperties;
+}
+
+export default function LineSeries(props: LineSeriesProps) {
+  const { data, label, ...otherProps } = props;
+  // Update plot context with data description
+  const { dispatch } = useDispatchContext();
+  useEffect(() => {
+    const [xMin, xMax] = extent(data, (d) => d.x);
+    const [yMin, yMax] = extent(data, (d) => d.y);
+    dispatch({ type: 'newData', value: { xMin, xMax, yMin, yMax, label } });
+  }, [data, label, dispatch]);
+
+  // Render stateless plot component
+  return <LineSeriesRender {...otherProps} data={data} />;
+}
+
+function LineSeriesRender({
+  data,
+  lineStyle,
+  width,
+  height,
+}: LineSeriesRenderProps) {
+  // Get scales from context
+  const { xMin, xMax, yMin, yMax } = usePlotContext();
+  if ([xMin, xMax, yMin, yMax].includes(undefined)) return null;
+
+  const xScale = scaleLinear().range([0, width]).domain([xMin, xMax]);
+  const yScale = scaleLinear().range([height, 0]).domain([yMin, yMax]);
```

Comment:
You can create the scales in Plot and pass them to the context. That way you won't need to pass width and height to this component.